### PR TITLE
Updated SSG JRE content to match current STIGs

### PR DESCRIPTION
**Pull request by cyarbrough76**
#### Description:

- _Combined checks and remediations where covered by a single STIG. Added new checks and remediations for new STIGs. Updated profile and overlay._

#### Rationale:

- _Updated JRE content to match current STIG requirements_

### CODE REVIEW

**Comment by mildas** [comment]
**Changes identified:**
Rule jre:
 The rule doesn't occur in any profile nor product.
 Bash remediation was deleted. No test for it will be selected.
 Bash remediation is newly added.
 Found change in bash remediation.
Rule java_jre_askgrantdialog_notinca:
 OVAL check is newly added.
Rule java_jre_askgrantdialog_show:
 OVAL check is newly added.
Rule java_jre_blacklist:
 OVAL check is newly added.
Rule java_jre_configure_crypto_policy:
 The rule doesn't occur in any profile nor product.
 Node moved within OVAL check.
 Node deleted from OVAL check.
 New node inserted to OVAL check.
 Text added outsite tags in OVAL check.
Rule java_jre_deployment_config_mandatory:
 OVAL check for java_jre_deployment_config_mandatory was deleted.
Rule java_jre_deployment_config_properties:
 Text changed in OVAL check.
 New node inserted to OVAL check.
Rule java_jre_exception_sites_enabled:
 OVAL check is newly added.
Rule java_jre_exception_sites_exists:
 OVAL check is newly added.
Rule java_jre_insecure_jres:
 OVAL check is newly added.
Rule java_jre_revocation_check:
 OVAL check is newly added.
Rule java_jre_security_level:
 OVAL check is newly added.
Rule java_jre_untrusted_sources:
 OVAL check for java_jre_untrusted_sources was deleted.
Rule java_jre_untrusted_sources_locked:
 OVAL check for java_jre_untrusted_sources_locked was deleted.
Rule java_jre_validation_crl:
 Text changed in OVAL check.
 New node inserted to OVAL check.
Rule java_jre_validation_crl_locked:
 OVAL check for java_jre_validation_crl_locked was deleted.
Rule java_jre_validation_ocsp:
 Text changed in OVAL check.
 New node inserted to OVAL check.
Rule java_jre_validation_ocsp_locked:
 OVAL check for java_jre_validation_ocsp_locked was deleted.
Rule java_jre_webjava:
 OVAL check is newly added.
Profile stig on jre:
 Rule java_jre_askgrantdialog_show, java_jre_exception_sites_exists, java_jre_askgrantdialog_notinca, java_jre_security_level, java_jre_blacklist, java_jre_exception_sites_enabled, java_jre_revocation_check, java_jre_webjava, java_jre_previous_versions, java_jre_insecure_jres added to stig profile.
 Rule java_jre_untrusted_sources_locked, java_jre_deployment_config_mandatory, java_jre_validation_crl_locked, java_jre_validation_ocsp_locked, java_jre_untrusted_sources removed from stig profile.

**Recommended tests to execute:**
 build_product jre
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_askgrantdialog_notinca
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_askgrantdialog_show
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_blacklist
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_deployment_config_properties
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_exception_sites_enabled
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_exception_sites_exists
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_insecure_jres
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_revocation_check
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_security_level
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_validation_crl
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_validation_ocsp
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_webjava
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-jre-ds.xml stig

File: jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml [hunk 1]
```diff
@@ -0,0 +1,34 @@
+<def-group>
+  <definition class="compliance" id="java_jre_askgrantdialog_notinca" version="1">
+    <metadata>
+      <title>Disable Java Execution From Untrusted Sources</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml:4]: The rule title should match this title.

File: jre/guide/java/java_jre_askgrantdialog_show/oval/jre.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
+<def-group>
+  <definition class="compliance" id="java_jre_askgrantdialog_show" version="1">
+    <metadata>
+      <title>Disable User Access to Untrusted Sources Setting</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_askgrantdialog_show/oval/jre.xml:4]: The rule title should match this title.

File: jre/guide/java/java_jre_blacklist/oval/jre.xml [hunk 3]
```diff
@@ -0,0 +1,35 @@
+<def-group>
+  <definition class="compliance" id="java_jre_blacklist" version="1">
+    <metadata>
+      <title>Prevent the Download of Prohibited Mobile Code.</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_blacklist/oval/jre.xml:4]: The rule title should match this title.

File: jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml [hunk 4]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-title: 'The Java Configuration File Exists'
+title: 'Oracle JRE 8 must have a deployment.config file present'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml:3]: This change doesn't need to take place.

File: jre/guide/java/java_jre_deployment_properties_exists/rule.yml [hunk 5]
```diff
@@ -1,6 +1,6 @@
 documentation_complete: true
 
-title: 'The Java Properties File Exists'
+title: 'Oracle JRE 8 must have a deployment.properties file present'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_deployment_properties_exists/rule.yml:3]: This title change needs to take place.

File: jre/guide/java/java_jre_exception_sites_exists/oval/jre.xml [hunk 6]
```diff
@@ -0,0 +1,21 @@
+<def-group>
+  <definition class="compliance" id="java_jre_exception_sites_exists" version="1">
+    <metadata>
+      <title>The exception.sites File Exists</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_exception_sites_exists/oval/jre.xml:4]: The rule title should match this title.

File: jre/guide/java/java_jre_exception_sites_enabled/rule.yml [hunk 7]
```diff
@@ -0,0 +1,28 @@
+documentation_complete: true
+
+title: 'Oracle JRE 8 must enable the option to use an accepted sites list'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_exception_sites_enabled/rule.yml:3]: The rule title should match this title.

File: jre/guide/java/java_jre_insecure_jres/oval/jre.xml [hunk 8]
```diff
@@ -0,0 +1,34 @@
+<def-group>
+  <definition class="compliance" id="java_jre_insecure_jres" version="1">
+    <metadata>
+      <title>Prompt For Execution</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_insecure_jres/oval/jre.xml:4]: Rule title should match this title.

File: jre/guide/java/java_jre_revocation_check/oval/jre.xml [hunk 9]
```diff
@@ -0,0 +1,34 @@
+<def-group>
+  <definition class="compliance" id="java_jre_revocation_check" version="1">
+    <metadata>
+      <title>Check All Certificates</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_revocation_check/oval/jre.xml:4]: This title needs to be updated to something like `Force Certificate Revocation Checking`

Comment (redhatrises) [reply jre/guide/java/java_jre_revocation_check/oval/jre.xml:4]: The rule title should also match this updated title.

File: jre/guide/java/java_jre_security_level/oval/jre.xml [hunk 10]
```diff
@@ -0,0 +1,35 @@
+<def-group>
+  <definition class="compliance" id="java_jre_security_level" version="1">
+    <metadata>
+      <title>Enable Maximum Default Security</title>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_security_level/oval/jre.xml:4]: Rule title should match this title.

File: jre/guide/java/java_jre_validation_crl_locked/rule.yml [hunk 11]
```diff
@@ -1,29 +0,0 @@
-documentation_complete: true
-
-title: 'Disable User Access to Revocation Check Settings'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_validation_crl_locked/rule.yml:3]: This content should not be removed.

File: jre/guide/java/java_jre_untrusted_sources/rule.yml [hunk 12]
```diff
@@ -1,28 +0,0 @@
-documentation_complete: true
-
-title: 'Disable Java Execution From Untrusted Sources'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_untrusted_sources/rule.yml:3]: This content should not be removed.

File: jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_mandatory/rule.yml [hunk 13]
```diff
@@ -1,30 +0,0 @@
-documentation_complete: true
-
-title: 'Configure The Java Deployment Mandatory Setting'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_mandatory/rule.yml:3]: This content should not be removed.

File: jre/guide/java/java_jre_untrusted_sources_locked/rule.yml [hunk 14]
```diff
@@ -1,30 +0,0 @@
-documentation_complete: true
-
-title: 'Disable User Access to Untrusted Sources Settings'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_untrusted_sources_locked/rule.yml:3]: This content should not be removed.

File: jre/guide/java/java_jre_validation_ocsp_locked/rule.yml [hunk 15]
```diff
@@ -1,30 +0,0 @@
-documentation_complete: true
-
-title: 'Disable User Access to Online Certificate Validation Settings'
```
Comment (redhatrises) [inline jre/guide/java/java_jre_validation_ocsp_locked/rule.yml:3]: This content should not be removed.

File: jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml [hunk 16]
```diff
@@ -0,0 +1,34 @@
...
+    <ind:pattern operation="pattern match">^deployment.security.askgrantdialog.notinca=false$</ind:pattern>
+    <ind:instance datatype="int">1</ind:instance>
+  </ind:textfilecontent54_object>
+
+  <ind:textfilecontent54_test check="all" comment="Check deployment.security.askgrantdialog.notinca.lock" id="test_java_jre_askgrantdialog_notinca_locked" version="1">
+    <ind:object object_ref="object_java_jre_askgrantdialog_notinca_locked" />
+  </ind:textfilecontent54_test>
+  <ind:textfilecontent54_object id="object_java_jre_askgrantdialog_notinca_locked" version="1">
+    <ind:filepath>/etc/.java/deployment/deployment.properties</ind:filepath>
+    <ind:pattern operation="pattern match">^deployment.security.askgrantdialog.notinca.locked$</ind:pattern>
```
Comment (redhatrises) [inline jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml:31]: As this is a separate configuration item, it should really be in its own rule. So, either we deal with this now, or next month when we plan to update the content for multiple products/versions as well as will be submitting to DISA.

**Comment by cyarbrough76** [comment]
Will work on correcting the above items.

**Comment by matejak** [comment]
test this please

**Comment by mildas** [comment]
**Changes identified:**
Rule jre:
 Found change in bash remediation.
 The rule doesn't occur in any profile nor product.
 Bash remediation was deleted. No test for it will be selected.
 Bash remediation is newly added.
Rule java_jre_askgrantdialog_notinca:
 OVAL check is newly added.
Rule java_jre_askgrantdialog_show:
 OVAL check is newly added.
Rule java_jre_blacklist:
 OVAL check is newly added.
Rule java_jre_configure_crypto_policy:
 Node deleted from OVAL check.
 Node moved within OVAL check.
 New node inserted to OVAL check.
 The rule doesn't occur in any profile nor product.
 Text added outsite tags in OVAL check.
Rule java_jre_deployment_config_mandatory:
 OVAL check for java_jre_deployment_config_mandatory was deleted.
Rule java_jre_deployment_config_properties:
 Text changed in OVAL check.
 New node inserted to OVAL check.
Rule java_jre_exception_sites_enabled:
 OVAL check is newly added.
Rule java_jre_exception_sites_exists:
 OVAL check is newly added.
Rule java_jre_insecure_jres:
 OVAL check is newly added.
Rule java_jre_revocation_check:
 OVAL check is newly added.
Rule java_jre_security_level:
 OVAL check is newly added.
Rule java_jre_untrusted_sources:
 OVAL check for java_jre_untrusted_sources was deleted.
Rule java_jre_untrusted_sources_locked:
 OVAL check for java_jre_untrusted_sources_locked was deleted.
Rule java_jre_validation_crl:
 Text changed in OVAL check.
 New node inserted to OVAL check.
Rule java_jre_validation_crl_locked:
 OVAL check for java_jre_validation_crl_locked was deleted.
Rule java_jre_validation_ocsp:
 Text changed in OVAL check.
 New node inserted to OVAL check.
Rule java_jre_validation_ocsp_locked:
 OVAL check for java_jre_validation_ocsp_locked was deleted.
Rule java_jre_webjava:
 OVAL check is newly added.
Profile stig on jre:
 Rule java_jre_askgrantdialog_show, java_jre_blacklist, java_jre_exception_sites_enabled, java_jre_revocation_check, java_jre_webjava, java_jre_exception_sites_exists, java_jre_askgrantdialog_notinca, java_jre_previous_versions, java_jre_security_level, java_jre_insecure_jres added to stig profile.
 Rule java_jre_untrusted_sources, java_jre_deployment_config_mandatory, java_jre_validation_crl_locked, java_jre_validation_ocsp_locked, java_jre_untrusted_sources_locked removed from stig profile.

**Recommended tests to execute:**
 build_product jre
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_askgrantdialog_notinca
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_askgrantdialog_show
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_blacklist
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_deployment_config_properties
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_exception_sites_enabled
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_exception_sites_exists
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_insecure_jres
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_revocation_check
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_security_level
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_validation_crl
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_validation_ocsp
 tests/test_suite.py rule --libvirt qemu:///system test-suite-vm --remediate-using bash --datastream build/ssg-jre-ds.xml java_jre_webjava
 tests/test_suite.py profile --libvirt qemu:///system test-suite-vm --datastream build/ssg-jre-ds.xml stig

File: jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml (hunk 4 above)
Comment (thenefield) [reply jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml:3]: These titles were changes to align with the title in the STIG.

File: jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml (hunk 16 above)
Comment (thenefield) [reply jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml:31]: Can you provide an example of this? It is my understanding that each requirement should map to a single check and remediation. How do we reference multiple rules for a single requirement in the overlay? How would these then relate back to the STIG? If we were to import the results into the STIG checklists, the checks would no longer align with the requirements.

File: jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml (hunk 4 above)
Comment (redhatrises) [reply jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml:3]: I know you want to make a literal 1:1 to the STIG, but these type of changes are regressions as there are different types of java that these rules map to. In other words, this rule is mapped to IBM, Oracle, and OpenJDK making this change a bug.

**Comment by thenefield** [comment]
I am really not understanding the motive for the requested changes to these commits.

The only profile currently applicable to this content is the STIG.

The rule titles were updated to reflect what is indicated in the STIG. This provides a user the ability to easily map a rule title in the report with a rule title in the STIG.

The removed content no longer applies to the current STIG, so why would we keep it?

The request to split checks and remediations into different rules which would then no longer directly map back to a requirement doesn't make sense. I understand the argument that you want each configuration item individually mapped to their CCI. But in these cases, the rules you want to split are already mapped to the same CCI, per the STIG. So it kind of defeats the point of splitting them up if they both pertain to the same CCI.

The whole goal of this merge is to align the outdated content with the current STIG. What you are suggesting goes against that goal. The sole purpose of this content is to align with the STIG.

Please help me to understand your sense in direction for these changes.

File: jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml (hunk 4 above)
Comment (thenefield) [reply jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_exists/rule.yml:3]: This makes perfect sense. Thank you!

**Comment by redhatrises** [comment]
> I am really not understanding the motive for the requested changes to these commits.
> 
> The only profile currently applicable to this content is the STIG.
> 
> The rule titles were updated to reflect what is indicated in the STIG. This provides a user the ability to easily map a rule title in the report with a rule title in the STIG.
> 
> The removed content no longer applies to the current STIG, so why would we keep it?
> 
> The request to split checks and remediations into different rules which would then no longer directly map back to a requirement doesn't make sense. I understand the argument that you want each configuration item individually mapped to their CCI. But in these cases, the rules you want to split are already mapped to the same CCI, per the STIG. So it kind of defeats the point of splitting them up if they both pertain to the same CCI.
> 
> The whole goal of this merge is to align the outdated content with the current STIG. What you are suggesting goes against that goal. The sole purpose of this content is to align with the STIG.
> 
> Please help me to understand your sense in direction for these changes.

Just a reminder that this repo is more upstream and not a 1:1 relationship (word-for-word copy) with produced STIGs, so it will be different. It is from this very repo that vendor STIGs get created and submitted. Building flexibility in the rules makes the content become higher quality and allows them to be used for more than just a STIG itself. It also helps us help the ops folks tweak settings better according to their operational needs and requirements. Granted there has been a lack of... updated content (i.e. do we have the same rule coverage) for STIGs across the products and versions, but we are hopefully in the midst of changing that. And like I said, our plan is to turn around and submit a new jre STIG.

At the most, rules shouldn't be dropped at all if they still can be configured in the product. So, the choice is yours. Merge now minus the rule changes and have a massive update for java 8 and 11, or help make some of the updates now.

**Comment by matejak** [comment]
I would like to add some points:

- `<title>{{{ rule_title }}}</title>` in OVAL files will make sure that the OVAL title in the datastream is in sync with the rule title.
- The project is a security content library, so even if some rules don't make sense any more to your use case, we don't have to throw them out of the library. However, if conditions changed and the rule is not useful any more, we could take some action. The reason just needs to be communicated.
- The situation when a security policy mandates rule titles, but the XCCDF standard doesn't support per-profile rule titles is unfortunate. The mandated title may not be better than any other title, and that would also mean that two different policies that mandate different titles of one rule would cause problems in the repository. I see two possible outcomes:
  - Change those titles, as right now it is only the STIG regulation that imposes titles, and titles are not API, they are just informational.
  - Use jinja if block, e.g. `{{% if CUSTOMIZE_DESCRIPTIONS_TEXT = "STIG" %}}<STIG title>{{% else %}}<generic title>{{% endif %}}`. We can easily introduce a build-time configuration (e.g. `-DCUSTOMIZE_DESCRIPTIONS_TEXT=STIG`) that would propagate that variable to the Jinja substitution dictionary. The macro wouldn't break anything, it would simply pick the generic title until that functionality is implemented and someone uses it.

**Comment by thenefield** [comment]
I can certainly understand your viewpoints. Thank you.

My only question now is how do we split up the rules in such a way that we can still reference them in the STIG overlay? Is there an example you can provide?

So for instance, in the example above "java_jre_askgrantdialog_notinca".

The STIG Overlay currently indicates the following for this rule:

	<overlay owner="disastig" ruleid="java_jre_askgrantdialog_notinca" ownerid="JRE8-UX-000080" disa="1695" severity="medium">
		<VMSinfo VKey="66917" SVKey="81407" VRelease="1" />
		<title>Oracle JRE 8 must disable the dialog enabling users to grant permissions to execute signed content from an untrusted authority.</title>
	</overlay>

So if we now split this single requirement into two rules, one for the configuration, and the other to lock it, how do we indicate both rules into the overlay? Can we add more than one entry for the ruleid?

I just want to ensure what we are referencing is the complete solution and not just a part of it.

Thanks!

File: jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_properties/bash/jre.sh [hunk 17]
```diff
@@ -2,8 +2,8 @@
 JAVA_CONFIG="/etc/.java/deployment/deployment.config"
 JAVA_PROPERTIES="/etc/.java/deployment/deployment.properties"
 
-grep -q "^deployment.system.config=file://${JAVA_CONFIG}$" ${JAVA_CONFIG} && \
-sed -i "s;deployment.system.config=.*;deployment.system.config=file:\/\/${JAVA_PROPERTIES};g" ${JAVA_CONFIG}
-if ! [ $? -eq 0 ] ; then
-  echo "deployment.system.config=file://${JAVA_PROPERTIES}" >> ${JAVA_CONFIG}
-fi
+sed -i "/deployment.system.config=.*/d" ${JAVA_CONFIG}
```
Comment (redhatrises) [inline jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_properties/bash/jre.sh:5]: What are we making a change when we don't need to make a change? This seems like a regression and doesn't lead itself to idempotent behavior.

Comment (thenefield) [reply jre/guide/java/java_jre_deployment_config_configured/java_jre_deployment_config_properties/bash/jre.sh:5]: The previous change defined the property incorrectly. If a system is upgrading from the previous release, they would end up with two entries, one being correct and one not. The suggested change removes any previous entry and applies the correct entry.

We have to test both cases of a new install and upgrading from the previous release.

File: jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml (hunk 16 above)
Comment (ggbecker) [reply jre/guide/java/java_jre_askgrantdialog_notinca/oval/jre.xml:31]: @thenefield The ideal scenario would be to request a fix in the STIG itself from DISA. At this moment what we can do is just to split the configuration check into two and wait for an update to be able to assign a new STIG id.

**Comment by ggbecker** [comment]
> I can certainly understand your viewpoints. Thank you.
> 
> My only question now is how do we split up the rules in such a way that we can still reference them in the STIG overlay? Is there an example you can provide?
> 
> So for instance, in the example above "java_jre_askgrantdialog_notinca".
> 
> The STIG Overlay currently indicates the following for this rule:
> 
> ```
> <overlay owner="disastig" ruleid="java_jre_askgrantdialog_notinca" ownerid="JRE8-UX-000080" disa="1695" severity="medium">
> 	<VMSinfo VKey="66917" SVKey="81407" VRelease="1" />
> 	<title>Oracle JRE 8 must disable the dialog enabling users to grant permissions to execute signed content from an untrusted authority.</title>
> </overlay>
> ```
> 
> So if we now split this single requirement into two rules, one for the configuration, and the other to lock it, how do we indicate both rules into the overlay? Can we add more than one entry for the ruleid?
> 
> I just want to ensure what we are referencing is the complete solution and not just a part of it.
> 
> Thanks!

Unfortunately we cannot reference two rules with the same STIG id in the overlay. The correct approach is to request a fix from DISA in this particular STIG to split these configuration item into two so we have two different STIG ids to assign which will provide a one-to-one relation.

**Comment by redhatrises** [comment]
This work is being done in a different PR. Closing.

**Pull request closed without merging**